plotly_resampler/aggregation/aggregators.py

```diff
@@ -352,51 +318,53 @@ def __init__(
             Whether None values should be added when there are gaps / irregularly
             sampled data. A quantile-based approach is used to determine the gaps /
             irregularly sampled data. By default, True.
-        nan_position: str, optional
-            Indicates where nans must be placed when gaps are detected. \n
-            If ``'end'``, the first point after a gap will be replaced with a
-            nan-value \n
-            If ``'begin'``, the last point before a gap will be replaced with a
-            nan-value \n
-            If ``'both'``, both the encompassing gap datapoints are replaced with
-            nan-values \n
-            .. note::
-                This parameter only has an effect when ``interleave_gaps`` is set
-                to *True*.
-        dtype_regex_list: List[str], optional
-            List containing the regex matching the supported datatypes, by default None.
 
         """
         self.aggregation_func = aggregation_func
-        super().__init__(interleave_gaps, nan_position, dtype_regex_list)
-
-    def _aggregate(self, s: pd.Series, n_out: int) -> pd.Series:
-        if isinstance(s.index, pd.DatetimeIndex):
-            t_start, t_end = s.index[:: len(s) - 1]
-            rate = (t_end - t_start) / n_out
-            return s.resample(rate).apply(self.aggregation_func).dropna()
-
-        # no time index -> use the every nth heuristic
-        group_size = max(1, np.ceil(len(s) / n_out))
-        s_out = (
-            s.groupby(
-                # create an array of [0, 0, 0, ...., n_out, n_out]
-                # where each value is repeated based $len(s)/n_out$ times
-                by=np.repeat(np.arange(n_out), group_size)[: len(s)]
-            )
-            .agg(self.aggregation_func)
-            .dropna()
+        super().__init__(
+            interleave_gaps,
+            x_dtype_regex_list=x_dtype_regex_list,
+            y_dtype_regex_list=y_dtype_regex_list,
         )
+
+    def _aggregate(
+        self,
+        x: Optional[np.ndarray] = None,
+        y: Optional[np.ndarray] = None,
+        n_out: int = None,
+        **kwargs,
+    ) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
Are these the same input arrays? Let's be clear if it's an in-place modification.